### ISSUE TITLE
FR: Installing as not web root

### ISSUE THREAD

**Issue by gerroon** [wallabag/wallabag#1913]
### Issue details

Hi

Can you please provide a way to install v2 as a non web root app? At the moment the installation setup is done in a way that assumes that one will only have Wallabag as the only app on the server. I was able to run wallabag v1 as a normal server folder without further complications. I feel like v2 has made it complicated for users who want to host their version who also are not very high level web professionals.

Please bear in mind that not everyone is able to understand how Symphony or Ruby etc work. So I am hoping that the devs would consider making self hosting easier for the rest of us.

thanks 
### Environment
- wallabag version (or git revision) that exhibits the issue: 
  v2
- How did you install wallabag? Via `git clone` or by downloading the package?
  git clone 
- Last wallabag version that did not exhibit the issue (if applicable):
  v1
- php version:
- OS:
  linux 
- type of hosting (shared or dedicated):
  self
- which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL):
  sqlite
### Steps to reproduce/test case

Please provide necessary steps for reproduction of this issue, or better the
reduced test case (without any external dependencies, if possible).

**Comment by nicosomb** [comment]
Hello @gerroon! 

Can you try the **Installation on a shared hosting** process (here: https://www.wallabag.org/pages/download-wallabag.html) and tell me if it's OK for you please?

**Comment by gerroon** [comment]
@nicosomb 


Thanks for the follow up. I am sorry that I did not mention a critical information. I was trying to use it under apache and the current virtual hosting setup (the one you mentioned) assumes that it should be installed as the web root. I personally do not know how to not to affect my current apache setup with this new virtual host setup that is in the doc. If I go with that I wont be able to access my other web applications on my server.

I hope it makes sense.

**Comment by PeterTGarner** [comment]
> installation setup is done in a way that assumes that one will only have Wallabag as the only app on the server ... not sure if I can have multiple document roots under the same virtualhost (*:443 in my case)

I agree with Gerroon. I wouldn't say I'm a noob at setting up a web server, but I'm not an expert either. My specific issue is that my web server is https-only, so I don't have a wildcard certificate for subdomains. Is there a way to make v2 addressable in the same as v1? 

Wallabag is a great piece of software and I use it constantly - thank you for your work.

**Environment**

* wallabag version (or git revision) that exhibits the issue: 
v2.0.3

* How did you install wallabag? Via git clone or by downloading the package?
git clone

* Last wallabag version that did not exhibit the issue (if applicable):
v1.9,2

* php version:
 5.6.11-1ubuntu3.2

* OS:
Linux odroid1 3.10.92 #1 SMP PREEMPT Tue Nov 17 00:15:24 BRST 2015 armv7l

* type of hosting (shared or dedicated):
self - dedicated

* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL):
sqlite

* web server
nginx 1.9.3 (ubuntu)

**Comment by gerroon** [comment]
Any ideas?

**Comment by Dalan94** [comment]
I agree with all of you. It's really too complicated to install wallabag, especially when there is other software installed on the server. I tried to install it and after one hour I decided to come back with the old, but functioning, 1.9.1 version. 
Is it possible to install it like the 1 version (and the majority of self-hosted software like owncloud/freshrss) ?

**Comment by Poorchop** [comment]
Yes it is still possible to install it like you would install 1.9.1. As @nicosomb suggested, you would just follow the instructions for installing on shared hosting. The latest-v2-package already includes all of the dependencies that you need in order to simply unzip the file and start using wallabag. As long as you have configured PHP properly and you have installed all of the necessary extensions, then installation is almost exactly the same as v1.

Just unpack the archive, move it to `/var/www` or wherever you wish to serve the files, and use the preconfigured login information to start using wallabag. Any additional files and folders like configs, logs, and cache are automatically generated the first time that you access wallabag through your web browser.

**Comment by kleemo** [comment]
I have the same issue. It's not possible to install it like 1.9.1 or please put up very clear information. what does it mean to move the unpacked archive to "/var/www or wherever you wish to serve the files". what is the address that browser should point to? it's not working for me. can you please write a step by step? the documentation stating the virtual host for shared server is not clear at all. i would simply like to install the app on a subdomain or a subfolder of my shared server. thank you!

**Comment by Poorchop** [comment]
The step-by-step instructions are already provided in the [documentation](http://doc.wallabag.org/en/master/user/installation.html). It has been a long time since I installed v1 but I remember it being exactly the same apart from the fact that v1 had a web interface that checked your PHP setup and gave you a warning if an extension was missing. Now Composer is supposed to handle that but you don't need to use Composer at all if you just use the latest-v2-package method.

Install the PHP extensions listed at the top of installation documentation, then skip to the [shared hosting section](http://doc.wallabag.org/en/master/user/installation.html#on-a-shared-hosting).

>what is the address that browser should point to?

Exactly the same as wallabag v1 - it's determined by the $vhost.conf that you create yourself. Whether you're using Apache, nginx, or some other httpd, you point the root location to `/var/www/wallabag/web` and access it at the server name that you decide, such as wallabag.example.com. The docs already provide configurations for three httpds - Apache, nginx, and lighttpd. The last step is just setting proper ownership of the wallabag folder, which you also had to do with v1.

The only extra step that you _might_ need to do is install and configure a mail transfer agent if you wish to create more accounts. However, the latest-v2-package already comes configured with a default account that you can start using right away. Just remember to change the password.

If you're stuck on a step, then just ask on gitter. If I'm around, I'll try to help, otherwise someone else will surely answer in time.

**Comment by kleemo** [comment]
the documentation presumes that for shared hosting you have to do a vhost. but most shared hosting don't allow that (at least mine). i've done the steps requested. but i can't do a vhost. so it's actually not like the v1 unfortunately (that btw could be installed via cpanel). i already asked the question on gitter. thanks for your help.

**Comment by Dalan94** [comment]
I try to extract the all in one package and access it with my browser https:://mydomain.com/wallabag or https:://mydomain.com/wallabag/web and I access to the file directly…
So no it's not like the old v1. The problem is probably that my vhost.conf is bad but I don't want to have one especially for wallabag. 
Furthermore if I need to configure a mail server to create more user I don't want to upgrade to v2 because I don't want a mail server.

**Comment by Poorchop** [comment]
I'm not really sure what you're asking. I needed to create a vhost.conf for v1 too when I installed it. It was the exact same process of downloading the package, extracting it to `/var/www`, and creating the httpd conf pointing to wallabag.

>https:://mydomain.com/wallabag or https:://mydomain.com/wallabag/web

You only need one colon and you set the root to the wallabag/web directory in your vhost.conf. Then you just access your wallabag at mydomain.com/wallabag, wallabag.mydomina.com, or whatever server name you set.

>Furthermore if I need to configure a mail server to create more user I don't want to upgrade to v2 because I don't want a mail server.

You don't need to configure an entire mail server, just a mail transfer agent. Also, didn't v1 only allow you to create a single user per installation? I thought the fact that v2 supports multiple accounts was a new feature. As I said, v2 already includes one default account and you can probably create additional users manually via command line if you wish, although I'm not sure if this is officially supported. Kleemo's issue is different since he's using shared hosting, but if you're using a dedicated server/VPS, installation using the latest-v2-package should be almost exactly the same as v1.

**Comment by kleemo** [comment]
i can't setup a vhost in my shared server. do i have other options?

**Comment by kleemo** [comment]
can i do the same config with a .htaccess file? if so, could you tell me what should be the script and which folder to place it? i already have a .htaccess file in my root folder for another app. would i place it in the subdomain file where i want wallabag to be installed? thanks.

**Comment by Poorchop** [comment]
Just a thought but could you do `php bin/console server:run --env=prod` in order to run PHP's built-in server? According to the docs, you should then be able to access wallabag over port 8000. I have never used shared hosting so I don't know if you have shell access at all, but otherwise I don't know how you'd install any custom webapps.

**Comment by kleemo** [comment]
I've installed many webapps on my shared server. I tried cubet board for instance (similar to wallabag but not so relevant for me). In order to do what you suggest I think I'd have to have a fixed ip for my web server, which i wouldn't have.

**Comment by flow7** [comment]
I agree with the initial poster - I love wallabag and had V1 installed on my shared server (dreamhost). When V2 came out I looked at the installation procedures. 

Now I've retried - at [lab.project22.org](http://lab.project22.org) - it works, somewhere in a subdirectory /web.

As a relative noob (who only copy&pastes commands) I find the installation too difficult and really wish it came easier deployable (eg. with htaccess files etc), so it would be more accessible for more.

**Comment by gerroon** [comment]
I am not using a separate port for wallabag because I I do not want to install it as the root of my webserver. 

People keep saying that it is the exact same procedure, but if so how come I can run wallabag1 as xxx.com/wallabag but not wallabag2. V2 does not like how I did the v1 setup

**Comment by kratenko** [comment]
Read about this project and wanted to try it - but as it stands, there is no feasible way to install it on shared hosting, at least not according to the installation documentation.

I do have ssh-access on my shared hosting, and that is not common for every shared hoster. But I cannot create any vhosts on with my provider, and so I see no way to install. Any installation instruction that includes 

> After reloading or restarting Apache, you should now be able to access wallabag at http://domain.tld.

is not usable on most shared hosting environments, sorry.

If there is a way to install it, without a vhost, please give a howto. If it's not possible, that's sad and would most likely block me from using this; but in that case please state under requirements, that you need that. Only giving requirements to some php-packages let's me think that this is a project I can just ftp-drop into an accessible directory and then use, like so many php-projects are.

Thanks,
kratenko

**Comment by tcitworld** [comment]
There's a section on how to use the shared hosting package [here](http://doc.wallabag.org/en/master/user/installation.html#on-a-shared-hosting).

**Comment by kratenko** [comment]
Yes, I read that passage. It ends with: now setup your vhost. My point is, that setting up a vhost is nothing you can normally do on a shared hosting env, so those instructions cannot really be used for installation on shared hosting.

**Comment by tcitworld** [comment]
Documentation is wrong. The provided `web/.htaccess` should do all the job, no need for extra configuration.

**Comment by kratenko** [comment]
Yes, it is absolutely possible! Without any vhosts. All I had to do was direct an address to the /web subfolder. (Thinking about it now, adding a subdomain on my shared hoster is most likely adding a vhost internally.) Thanks for your help, it worked!

**Comment by gerroon** [comment]
@kratenko 

Can you please tell us exactly how you set it up?

**Comment by kratenko** [comment]
So, this is how I got it to run on my provider:
The web-control interface of my provider allows me to set up subdomains to my domains, pointing them to directories I can specify. So I put the whole contents of the zip file to my web space as `wallabag`and created a new subdomain pointing to `wallabag/web`. That about did it.

If you do not have subdomains to set up with your provider, I think you might be able to get it running, by tweaking .htaccess files rewriting to the `/web/` subdirectory, but I did not try it out. I think you'd need to take care to forbid web-access to all the other directories.

**Comment by jeffguy77** [comment]
I agree 100%. I am using shared hosting and have no access to vhosts. It is impossible to install. I have been a Linux admin for over 10 years.

**Comment by nicosomb** [comment]
Hello, 

I have a website in `/var/www/mywebsite`. 
I tried to installed wallabag into `/var/www/mywebsite/wallabag`: it works. I can access to http://mywebsite/wallabag/web. 
I created a new vhost, with a DocumentRoot to the `web` folder (`DocumentRoot "/var/www/mywebsite/wallabag/web"`), I can access my wallabag installation via the ServerName. 

I don't understand what are your problems.

**Comment by j0k3r** [comment]
This issue is fairly old and there hasn't been much activity on it. Closing, but please re-open if it still occurs.

**Comment by callmetango** [comment]
@nicosomb You described a setup of a virtual host with the `DocumentRoot` set to `/var/www/mywebsite/wallabag/web`. Opening http&#58;//mywebsite/ in the browser may work fine.

What I and the others are after is a setup where Wallabag is not installed in the document root but in a sublocation like `/var/www/mywebsite/htdocs/wallabag` that is reachable via `http://mywebsite/wallabag`. A typical use case is shared hosting where you have no privileges to add virtual hosts but you have the privileges to create subdirectories and an `.htaccess` file. So you are able to create a `Location` alias for /wallabag to point to /wallabag/web.

In technical terms:
```
DocumentRoot set to: /var/www/mywebsite/htdocs
Wallabag installed to: /var/www/mywebsite/htdocs/wallabag
Location alias configuration in the webserver: /wallabag
```

Opening http://mywebsite/wallabag in the browser won't work out of the box as of version [2.2.1](https://github.com/wallabag/wallabag/releases/tag/2.2.1) since Symfony redirects to `http://mywebsite/login` instead of `http://mywebsite/wallabag/login`.

So yes this issue still occurs and yes please merge pull request https://github.com/wallabag/wallabag/pull/2181 to solve this issue as well as https://github.com/wallabag/wallabag/issues/2273.

Glad to answer if you have further questions regarding the use cases.

**Comment by Caliandroid** [comment]
Hi,

just wanted to tell you, that i was able to get wallabag 2 (latest version 2.0.8) working on a crappy shared hoster without touching any vhost , .htaccess  or root folder...

It is installed in a subdomain https://www.example.com/wallabag2/ and accessible via https://www.example.com/wallabag2/web  

To be able to sync with the android app, i had first to create a client id under developer settings (forwarding url = https://www.example.com/wallabag2/web )

I've tried it some month ago without success so i guess there have been made some improvements to the code.

**Comment by grOGH** [comment]
I am trying to run the wallabag docker container with an nginx reverse proxy in its own container, routing to wallabag from a subdirectory (/wallabag).

Please consider approving the merge request since wallabag always redirects to /login, breaking the configuration.

**Comment by eeggenberger** [comment]
I was able to install Wallabag to a subdirectory without problems. I'm using Ubuntu, but this should work on other distributions too. My Apache document root is /var/www/html but Wallabag is installed to /var/www/wallabag. I've just created a config file based on the vhost setup in the docs and placed it in /etc/apache2/conf-available/wallabag.conf

`Alias /wallabag "/var/www/wallabag/web/"
<Directory "/var/www/wallabag/web">
	AllowOverride None
	Order Allow,Deny
	Allow from All
	Require all granted

	<IfModule mod_rewrite.c>
		Options -MultiViews
		RewriteEngine On
		RewriteCond %{REQUEST_FILENAME} !-f
		RewriteRule ^(.*)$ app.php [QSA,L]
	</IfModule>
</Directory>

<Directory /var/www/wallabag/web/bundles>
	<IfModule mod_rewrite.c>
		RewriteEngine Off
	</IfModule>
</Directory>`

Afterwards I've just enabled it with `a2enconf wallabag` and restarted Apache. Now Wallabag is working under the URL http://mydomain/wallabag and I didn't notice any problems yet. Like this, Wallabag is available under the path /wallabag on all your vhosts. If you're not using Ubuntu, you should be able to add the configuration above to your existing vhost.

**Comment by Arthinsx** [comment]
Après avoir galérer pour réussir l'installation de cette version (je dirais que j'y ai passé 3 bonnes heures; tout ça pour me rendre compte que j'avais interdit l'utilisation des fichiers .htaccess sur mon serveur), j'ai enfin réussi à accéder à mon wallabag v2 :)

Sauf que c'est moche car je n'ai accès à aucune ressource (image, css, ...).

J'ai corrigé ce problème en ajoutant un alias supplémentaire (même si cela me semble une solution temporaire)

`Alias /wallassets /var/www/wallabag/web/wallassets/`

**Comment by Kdecherf** [comment]
@Arthinsx peux-tu t'assurer que le paramètre `domain_name` est correctement renseigné dans `parameters.yml` avec wallabag 2.3.1 ? Cela devrait t'éviter de devoir rajouter un alias sur le serveur web.

**Comment by fantasio31** [comment]
Arthinsx solutions worked fine for me too, with same initial issue.
Checked "parameters.yml" domain value is ok for me : http://<mydomain>/wallabag/web. But no impact on access to wallassets files.

Thanks @Arthinsx  for this workaround.

**Comment by Arthinsx** [comment]
@Kdecherf : alors, j'avais un peu oublié ce post, mais même avec le bon domain_name à jour dans le fichier parameter.yml, ça ne passe pas. (je suis bien en 2.3.1)

@fantasio31 : you're welcome.

**Comment by leolivier** [comment]
Hi, pretty old post but still the problem exist. I'm trying to install wallabag with docker behind traefik as a reverse proxy and I need it to be accessible through https://mydomain/wallabag/As, the first thing wallabag does is to redirect to /login instead of ./login, I can't get it working.
Is there a way to do this???

**Comment by std2main** [comment]
Yunohost support install wallabag in sub directory, here is how they configure, may it help.
https://github.com/YunoHost-Apps/wallabag2_ynh/blob/testing/conf/nginx.conf

**Comment by JialuZhang** [comment]
@eeggenberger 

```
<Directory /var/www/wallabag/web/bundles>
	<IfModule mod_rewrite.c>
		RewriteEngine Off
	</IfModule>
</Directory>
```
In your posted configuration, the line: "RewriteEngine Off" is a misconfiguration, and adding it to your system will not change any system behavior.
Why "RewriteEngine Off" is allowed by Apache is that, if you include multiple "RewriteRule" parameters in your configuration, then instead of commenting them all, you can explicitly using “RewriteEngine Off” to disable all "RewriteRule".

More importantly, the default value of “RewriteEngine" is already an "off", so adding “RewriteEngine Off" is quite unnecessary and it may cause confusion to users.

Since herein there is no "RewriteRule", deleting “RewriteEngine Off” would be ideal.

Related Apache source code snippet:
```
run_rewritemap_programs(server_rec *s , apr_pool_t *p){
if (conf->state == ENGINE_DISABLED) { // usage of "RewriteEngine"
  return APR_SUCCESS; // early return
rewritemap_program(...); // usage of "RewriteRule" 
}
```
